routing/turns_generator.cpp

```diff
@@ -806,5 +887,57 @@ void GetTurnDirection(Index const & index, TurnInfo & turnInfo, TurnItem & turn)
     return;
   }
 }
+
+size_t CheckUTurnOnRoute(vector<LoadedPathSegment> const & segments, size_t currentSegment, TurnItem & turn)
+{
+  size_t constexpr kUTurnLookAhead = 3;
+  double const kUTurnHeadingSensitivity = math::pi / 10.0;
+
+  ASSERT_GREATER(segments.size(), 1, ());
+  ASSERT_GREATER(currentSegment, 0, ());
```

Comment:
Because in this and other function we process the turn between the previous and the current segment. So we need a shift to get the previous segment.